src/taxonomy/view.js

```diff
@@ -7,6 +7,16 @@ const updateURL = async ( action, value, name ) => {
 	} else {
 		url.searchParams.delete( name );
 	}
+
+	// Remove pagination when performing a search
+	// for your specific pagination format: query-{id}-page
+	const searchParams = url.searchParams;
+	[...searchParams.keys()].forEach(param => {
+		if (param.match(/query-\d+-page/) || param === 'paged') {
+			searchParams.delete(param);
+		}
+	});
```

Comment:
I think it should be possible to scope this to the current query loop block as this will reset pagination for all query loops on the page, and they should be able to work independently. E.g. if `name === 's'` remove the `paged` param, if `name !== 's'`, remove `name.replace( '-s', '-page' )`.